test/parallel/test-queue-microtask.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+
+assert.strictEqual(typeof queueMicrotask, 'function');
+
+[
+  undefined,
+  null,
+  0,
+  'x = 5',
+].forEach((t) => {
+  assert.throws(() => {
+    queueMicrotask(t);
+  }, {
+    code: 'ERR_INVALID_ARG_TYPE',
+  });
+});
+
+{
+  let called = false;
+  queueMicrotask(() => {
+    called = true;
+  });
+  assert.strictEqual(called, false);
+}
+
+queueMicrotask(function() {
+  assert.deepEquals(arguments, []);
+}, 'x', 'y');
+
+{
+  const q = [];
+  Promise.resolve().then(() => q.push('a'));
+  queueMicrotask(() => q.push('b'));
+  Promise.reject().catch(() => q.push('c'));
+
+  queueMicrotask(() => {
+    assert.deepStrictEqual(q, ['a', 'b', 'c']);
+  });
+}
```

Comment:
Can you add a test that shows the relationship with `process.nextTick`  and `setImmediate`?

Reply:
setImmediate and nextTick are tested separately, and there is a PR open to change the semantics of how microtasks and timers interleave so I think it would be best to leave that alone from this side.

Reply:
https://github.com/nodejs/node/pull/22842